src/services/ui/src/components/RHF/TextDisplay.tsx

```diff
@@ -0,0 +1,49 @@
+import { cn } from "@/utils";
+import { Link } from "react-router-dom";
+import { RHFTextField } from "shared-types";
+
+interface RHFTextDisplayProps {
+  text: RHFTextField;
+}
+
+export const RHFTextDisplay = (props: RHFTextDisplayProps) => {
+  if (!Array.isArray(props.text)) return props.text;
+
+  return (
+    <>
+      {...props.text?.map((t) => {
+        if (typeof t === "string") return <>{t}</>;
+
+        switch (t?.type) {
+          case "br":
+            return (
+              <>
+                <br /> {t.text}
+              </>
+            );
+          case "brWrap":
+            return (
+              <>
+                <br /> {t.text} <br />
+              </>
+            );
+          case "bold":
+            return <b className={t.classname}>{t.text}</b>;
+          case "italic":
+            return <i className={t.classname}>{t.text}</i>;
+          case "link":
+            return (
+              <Link
+                to={t.link ?? "/"}
+                className={cn("cursor-pointer text-blue-600 ml-0", t.classname)}
+              >
+                {t.text}
+              </Link>
+            );
+          default:
+            return <span className={t.classname}>{t.text}</span>;
```

Comment:
Went with span as there are a few places this is used where putting a p tag could break it. Since you can't p tag inside a p tag, div is also an option here. But I can definitely add it as a quick case.